src/Instrumentation/Symfony/src/SymfonyInstrumentation.php

```diff
@@ -165,5 +167,160 @@ public static function register(): void
                 return $params;
             },
         );
+
+        /**
+         * Extract symfony event dispatcher known events and trace the controller
+         *
+         * Adapted from https://github.com/DataDog/dd-trace-php/blob/master/src/DDTrace/Integrations/Symfony/SymfonyIntegration.php
+         */
+        hook(
+            EventDispatcher::class,
+            'dispatch',
+            pre: static function (
+                EventDispatcher $dispatcher,
+                array $params,
+                string $class,
+                string $function,
+                ?string $filename,
+                ?int $lineno,
+            ) use ($instrumentation): array {
+                if (!isset($args[0])) {
+                    return $params;
+                }
+
+                if (\is_object($args[0])) {
+                    // dispatch($event, string $eventName = null)
+                    $event = $args[0];
+                    $eventName = isset($args[1]) && \is_string($args[1]) ? $args[1] : \get_class($event);
+                } elseif (\is_string($args[0])) {
+                    // dispatch($eventName, Event $event = null)
+                    $eventName = $args[0];
+                    $event = isset($args[1]) && \is_object($args[1]) ? $args[1] : null;
+                } else {
+                    return $params;
+                }
+
+                if ($eventName === 'kernel.controller' && \method_exists($event, 'getController')) {
+                    $controller = $event->getController();
+                    if (!($controller instanceof \Closure)) {
+                        if (\is_callable($controller, false, $controllerName) && $controllerName !== null) {
+                            if (\strpos($controllerName, '::') > 0) {
+                                list($class, $method) = \explode('::', $controllerName);
+                                if (isset($class, $method)) {
+                                    hook(
```

Comment:
I don't think this will work with modern runtimes, where one long-lived application might handle many requests.
Maybe it should just manually create and end span on Symfony events?

Reply:
I think I'd rather memoize the hooks already in place. - this hook in particular is for the controller callable